app/controllers/events_controller.rb

```diff
@@ -0,0 +1,33 @@
+class EventsController < ApplicationController
+  def index
+    @events = Event.all.order('created_at DESC')
+    @past = Event.all.past_events(Time.zone.now).order('date DESC')
+    @upcoming = Event.all.upcoming_events(Time.zone.now).order('date ASC')
```

Comment:
There is no need for `.all`.
```suggestion
    @past = Event.past_events(Time.zone.now).order('date DESC')
    @upcoming = Event.upcoming_events(Time.zone.now).order('date ASC')
```